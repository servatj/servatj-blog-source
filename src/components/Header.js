import React from 'react'
import {StaticQuery, graphql} from 'gatsby'
import {rhythm} from '../utils/typography'

// eslint-disable-next-line react/prop-types
const TitleAndDescription = ({data, profilePic}) => {
  const title = data.site.siteMetadata.title
  const description = data.site.siteMetadata.description

  return (
    <div
      style={{
        display: 'flex',
        // flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
      }}
    >
      <div style={{marginBottom: -5, marginRight: 10}}>
        <img
          src={profilePic}
          alt={`Josep Servat`}
          style={{
            marginBottom: 0,
            width: rhythm(2),
            height: rhythm(2),
            borderRadius: '50%',
          }}
        />
      </div>

      <div>
        <h1 style={{marginBottom: 0}}>{title}</h1>
        <p
          style={{
            marginTop: 0,
            opacity: 0.5,
            color: '#ff6b6b',
          }}
        >
          {description}
        </p>
      </div>
    </div>
  )
}

const Header = () => {
  return (
    <StaticQuery
      query={graphql`
        query {
          site {
            siteMetadata {
              title
              description
            }
          }
        }
      `}
      render={data => (
        <TitleAndDescription
          data={data}
          profilePic={'../assets/josepServat.jpg'}
        />
      )}
    />
  )
}

export default Header
