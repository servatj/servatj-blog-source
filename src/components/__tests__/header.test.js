/* eslint-disable no-undef */
import React from 'react'
import renderer from 'react-test-renderer'
import {StaticQuery} from 'gatsby'
import Header from '../Header'

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({render}) =>
    render({
      site: {
        siteMetadata: {
          title: `Default Starter`,
        },
      },
    }),
  )
})

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Header />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
