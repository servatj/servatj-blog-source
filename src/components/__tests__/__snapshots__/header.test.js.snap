// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "center",
    }
  }
>
  <div
    style={
      Object {
        "marginBottom": -5,
        "marginRight": 10,
      }
    }
  >
    <img
      alt="Josep Servat"
      src="test-file-stub"
      style={
        Object {
          "borderRadius": "50%",
          "height": "3.5rem",
          "marginBottom": 0,
          "width": "3.5rem",
        }
      }
    />
  </div>
  <div>
    <h1
      style={
        Object {
          "marginBottom": 0,
        }
      }
    >
      Default Starter
    </h1>
    <p
      style={
        Object {
          "color": "#ff6b6b",
          "marginTop": 0,
          "opacity": 0.5,
        }
      }
    />
  </div>
</div>
`;
